internal/manifest/parse.go

```diff
@@ -115,3 +146,60 @@ func pullLayer(ctx context.Context, imageRef string, keyChain authn.Keychain) (c
 func getFsChartPath(imageSpec v1beta2.ImageSpec) string {
 	return filepath.Join(os.TempDir(), fmt.Sprintf("%s-%s", imageSpec.Name, imageSpec.Ref))
 }
+
+func untarLayer(tarPath string) (string, error) {
+	tarFile, err := os.Open(tarPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to open file: %w", err)
+	}
+	defer tarFile.Close()
+
+	tarReader := tar.NewReader(tarFile)
+
+	for {
+		header, err := tarReader.Next()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return "", fmt.Errorf("failed to read tar: %w", err)
+		}
+
+		if strings.HasPrefix(header.Name, "._") {
+			continue
+		}
+
+		extractedFilePath, err := sanitizeArchivePath(filepath.Dir(tarPath), header.Name)
+		if err != nil {
+			return "", fmt.Errorf("failed to sanitize archive path: %w", err)
+		}
+
+		if _, err := os.Stat(extractedFilePath); err == nil {
+			return extractedFilePath, nil
+		}
+
+		outFile, err := os.Create(extractedFilePath)
```

Comment:
I think here you'd better check the file type, if it's a folder, should return with error. check the switch case here: https://medium.com/@skdomino/taring-untaring-files-in-go-6b07cf56bc07

```
// check the file type
		switch header.Typeflag {
```